### PR TITLE
Create biz profile put endpoint

### PR DESCRIPTION
**Pull request by ruaraya**
Create new biz profile put endpoint. The modifications include the following:

- Creation of new service to update biz profile, including branch addresses
- Definition of related unit tests and integrated tests
- Correction of biz command objects
- Improve verification of existing integrated test cases
- Fix failed unit test case
- Minor fixes
- General refactoring and formatting

### CODE REVIEW

File: src/Spd.Manager.Licence/BizProfileValidation.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using FluentValidation;
+
+namespace Spd.Manager.Licence;
+public class BizProfileUpdateRequestValidator : AbstractValidator<BizProfileUpdateRequest>
+{
+    public BizProfileUpdateRequestValidator()
+    {
+        RuleFor(r => r.BizTradeName).NotEmpty();
+        RuleFor(r => r.BizTypeCode).NotEmpty().IsInEnum();
+        RuleFor(r => r.BizBCAddress).NotEmpty();
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/BizProfileValidation.cs:10]: it is not the rule, the rule is if BizAddress is not in BC, then BizBCAddress must exist

File: src/Spd.Manager.Licence/BizProfileValidation.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        RuleFor(r => r.BizMailingAddress)
+            .NotEmpty()
+            .When(r => r.MailingAddressIsSameBizAddress == false);
+        RuleFor(r => r)
+            .Must(r => r.BizMailingAddress?.AddressLine1 != r.BizBCAddress?.AddressLine1)
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/BizProfileValidation.cs:15]: what's this? it is possible that address line 1 is the same even though MailingAddressIsSameBizAddress = false.

File: src/Spd.Manager.Licence/BizProfileManager.cs [hunk 3]
```diff
@@ -182,4 +196,28 @@ private async Task<PortalUserResp> AddPortalUserToBiz(BceidIdentityInfo info, Gu
...
+        await _addressRepository.UpdateAddressesAsync(updateAddressCmd, ct);
+
+        // Create branches
+        List<BranchAddr> addressesToCreate = branchesToProcess.Where(b => b.BranchId == Guid.Empty || b.BranchId == null).ToList();
+        UpsertAddressCmd createAddressCmd = new()
+        {
+            BizId = bizId,
+            Addresses = addressesToCreate
+        };
+        await _addressRepository.CreateAddressesAsync(createAddressCmd, ct);
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/BizProfileManager.cs:221]: all this needs 3 transactions. I will create an improvement ticket as Update an array. All delete, update and create are done in the repository layer with one transaction.
It should apply to both Alias and Address.

File: src/Spd.Manager.Licence/BizProfileValidation.cs (hunk 1 above)
Comment (ruaraya) [reply src/Spd.Manager.Licence/BizProfileValidation.cs]: Fixed.

File: src/Spd.Manager.Licence/BizProfileValidation.cs (hunk 2 above)
Comment (ruaraya) [reply src/Spd.Manager.Licence/BizProfileValidation.cs]: I removed the flag "MailingAddressIsSameBizAddress" and adjusted the way the validation is done.

**Review by peggy-quartech — APPROVED** [review]

**Pull request merged**